tests/common/plugins/sanity_check/__init__.py

```diff
@@ -119,8 +130,51 @@ def do_checks(request, check_items, *args, **kwargs):
     return check_results
 
 
+@pytest.fixture(scope="module", autouse=True)
+def log_custom_msg(request):
+    yield
+    module_name = request.node.name
+    items = request.session.items
+    for item in items:
+        if item.module.__name__ + ".py" == module_name.split("/")[-1]:
+            customMsgDict = {}
+            dutChekResults = {}
+            for key in DUT_CHEK_LIST:
+                if request.config.cache.get(key, None) is False:
+                    dutChekResults[key] = False
+            if dutChekResults:
+                customMsgDict['DutChekResult'] = dutChekResults
+
+            # Check pre_sanity_checks results
+            preSanityCheckResults = {}
+            if request.config.cache.get("pre_sanity_check_failed", None):
+                preSanityCheckResults['pre_sanity_check_failed'] = True
+            # pre_sanity_recovered should be None in healthy case, record either True/False
+            if request.config.cache.get("pre_sanity_recovered", None) is not None:
+                preSanityCheckResults['pre_sanity_recovered'] = request.config.cache.get("pre_sanity_recovered")
+            if preSanityCheckResults:
+                customMsgDict['PreSanityCheckResults'] = preSanityCheckResults
+
+            # Check post_sanity_checks results
+            postSanityCheckResults = {}
+            if request.config.cache.get("post_sanity_check_failed", None):
+                postSanityCheckResults['post_sanity_check_failed'] = True
+            # post_sanity_recovered should be None in healthy case, record either True/False
+            if request.config.cache.get("post_sanity_recovered", None) is not None:
+                preSanityCheckResults['post_sanity_recovered'] = request.config.cache.get("post_sanity_recovered")
+            if postSanityCheckResults:
+                customMsgDict['PostSanityCheckResults'] = postSanityCheckResults
+
+            # if we have any custom message to log, append it to user_properties
+            if customMsgDict:
+                logger.debug("customMsgDict: {}".format(customMsgDict))
+                item.user_properties.append(('CustomMsg', json.dumps(customMsgDict)))
+
+    reset_cache_list(request)
+
+
 @pytest.fixture(scope="module")
-def sanity_check_full(localhost, duthosts, request, fanouthosts, nbrhosts, tbinfo):
+def sanity_check_full(localhost, duthosts, request, fanouthosts, nbrhosts, tbinfo, log_custom_msg):
```

Comment:
Having `log_custom_msg` in `sanity_check_full` will be a bit dangerous because we have multiple early returns & exception throwings in `sanity_check_full`, so it will be hard for us to maintain the config

Reply:
re-located into upper function, sanity_check, so log_custom_msg can get executed even when sanity_check option is off